specification.md

```diff
@@ -1859,9 +1860,21 @@ The `version` property can be used to provide a specific workflow version. It mu
 The `annotations` property defines a list of helpful terms describing the workflows intended purpose, subject areas, or other important qualities,
 for example "machine learning", "monitoring", "networking", etc
 
-The `dataInputSchema` property can be used to validate the workflow data input against a defined JSON Schema.
-This check should be done before any states are executed. `dataInputSchema` can have two different types.
-If it is an object type it has the following definition:
+The `dataInputSchema` and `dataOutputSchema` properties can be used to validate model against a defined JSON Schema.
+
+The `dataInputSchema` property validates the [workflow data input](#Workflow-Data-Input). Validation should be performed before any states are executed. In case of
+a starting [Event state](#Event-state), it is not used to validate its event payloads. The `failOnValidationErrors` property  determines if workflow execution should continue in case of validation errors. 
+
+The `dataOutputSchema` property validates the [Workflow data output](#workflow-data-output). Validation should be performed after workflow execution has finished successfully. Successfully means the workflow has completed an end state without errors. 
```

Comment:
I would instead write something like:

Successfully means that the workflow has completed an end state without errors